messaging/src/main/java/org/frankframework/extensions/mqtt/MqttFacade.java

```diff
@@ -85,14 +87,21 @@ public void configure() throws ConfigurationException {
 			connectOptions.setPassword(credentialFactory.getPassword().toCharArray());
 		}
 
-		MqttDefaultFilePersistence dataStore = new MqttDefaultFilePersistence(getPersistenceDirectory());
 		try {
-			client = new MqttClient(brokerUrl, clientId, dataStore);
+			client = new MqttClient(brokerUrl, clientId, getDataStore());
 		} catch (MqttException e) {
 			throw new ConfigurationException("Could not create client", e);
 		}
 	}
 
+	private MqttClientPersistence getDataStore() {
+		if (StringUtils.isEmpty(getPersistenceDirectory())) {
+			return new MemoryPersistence();
+		}
+
+		return new MqttDefaultFilePersistence(getPersistenceDirectory());
```

Comment:
Misschien dat je zelfs `TemporaryDirectoryUtils.getTempDirectory` oid kan gebruiken en dan de (schone) naam van de listener.

Reply:
Die directory wordt gebruikt om messages op te slaan zodat de client weet wat al verstuurd/ontvangen is. Dit gebruik je in een scenario waarbij je 100% zeker wilt zijn dat alles successvol verstuurt is (na een acknowledgement). Dit moet dus wel onder de controle van de gebruiker blijven